### PR TITLE
Fix multi delete data not effect

### PR DESCRIPTION
**Pull request by godchen0212**
Fix multi delete data not effect
Issue #7130
/kind enhancement

Signed-off-by: godchen <qingxiang.chen@zilliz.com>

### CODE REVIEW

**Review by XuanYang-cn — COMMENTED** [review]
/lgtm

File: internal/datanode/compactor.go [hunk 1]
```diff
@@ -97,9 +97,11 @@ func (t *compactionTask) mergeDeltalogs(dBlobs map[UniqueID][]*Blob, timetravelT
 	var (
 		pk2ts = make(map[UniqueID]Timestamp)
 		dbuff = &DelDataBuf{
-			delData: &DeleteData{Data: make(map[UniqueID]UniqueID)},
-			tsFrom:  math.MaxUint64,
-			tsTo:    0,
+			delData: &DeleteData{
+				Pks: make([]UniqueID, 0),
+				Tss: make([]Timestamp, 0)},
+			tsFrom: math.MaxUint64,
```
Comment (czs007) [inline internal/datanode/compactor.go:103]: in next pr, please change to use typeutil.MaxTimestamp and typeutil.ZeroTimestamp

**Pull request merged**